nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -898,6 +972,10 @@ ngx_http_modsecurity_handler(ngx_http_request_t *r)
     if (ctx == NULL) {
         ctx = ngx_http_modsecurity_create_ctx(r);
         if (ctx == NULL) {
+            modsec_latency_ret = ngx_http_calculate_modsec_latency(r, start_time);
+            if (modsec_latency_ret != NGX_OK) {
```

Comment:
Won't this eat up the InternalServerError? The method here is returning http status code, but in this case we seem to be returning nginx's standard return values of error/declined etc.
Calculation of the latency should be considered as secondary and we shouldn't change the actual flow. I would recommend logging an error instead of changing the return codes. But I guess it would still have to be validated what will happen when the variable is accessed in logging phase.

Reply:
Ah, it seems mixing NGX_ERROR, NGX_OK and NGX_HTTP_INTERNAL_SERVER_ERROR happens elsewhere as well. So it might not be that off of a thing to do. However, I think the refactoring suggested above should be considered.

Reply:
Refactored the code. We are only returning error in case memory allocation fails in which case we should send back an error and log it also.